sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -92,6 +105,124 @@ export class AssessmentSequentialProcessingService {
     return applicationRepo.save(impactedApplication);
   }
 
+  /**
+   * Finds all applications before the one related to the {@link assessmentId} provided
+   * and returns the sum of all awards associated with non-overwritten applications.
+   * Only pending awards or already sent awards will be considered.
+   * Only federal and provincial grants are considered.
+   * Grants that are common between part-time and full-time (e.g. CSGP, SBSD) applications will have
+   * their totals calculated per intensity, which means that if the student has a part-time
+   * and a full-time application in the same year and both have a CSGP grant, its totals will
+   * be individually calculated and returned.
+   * The chronology of the applications is defined by the method {@link getSequencedApplications}.
+   * @param assessmentId assessment id to be used as a reference to find the past applications.
+   * @param options method options.
+   * - `alternativeReferenceDate` date that should be used to determine the order when the
+   * {@link assessmentId} used as reference does not have a calculated date yet.
+   * @returns list of existing awards and their totals, if any, otherwise it returns an empty array,
+   * for instance, if the application is the first application or the only application for the
+   * program year.
+   */
+  async getProgramYearPreviousAwardsTotals(
+    assessmentId: number,
+    options?: { alternativeReferenceDate?: Date },
+  ): Promise<AwardTotal[]> {
+    const assessment = await this.studentAssessmentRepo.findOne({
+      select: {
+        id: true,
+        application: {
+          id: true,
+          applicationNumber: true,
+          student: {
+            id: true,
+          },
+          programYear: {
+            id: true,
+          },
+        },
+      },
+      relations: {
+        application: { student: true, programYear: true },
+      },
+      where: {
+        id: assessmentId,
+      },
+    });
+    const sequencedApplications = await this.getSequencedApplications(
+      assessment.application.applicationNumber,
+      assessment.application.student.id,
+      assessment.application.programYear.id,
+      { alternativeReferenceDate: options?.alternativeReferenceDate },
+    );
+    if (!sequencedApplications.previous.length) {
+      // There are no past applications.
+      return [];
+    }
+    const applicationNumbers = sequencedApplications.previous.map(
+      (application) => application.applicationNumber,
+    );
+    const totals = await this.studentAssessmentRepo
```

Comment:
will it be easy to go from application repo considering only current assessment ??

Reply:
Just double-checking the understanding here, any disbursement sent associated with the current application assessment must be considered. Does it make sense?

Reply:
After our chat, I will change the query to work based on the current assessment.

Reply:
Query updated to consider the awards only from the current assessment.